tools/GulpUtils.ts

```diff
@@ -246,6 +198,151 @@ class GulpUtils {
             return folder;
         }, start + path.sep);
     }
+
+    private static updateInternalDependencies(srcPath: string, tacoModules: string[], buildType: string, destPath?: string): Q.Promise<any> {
+        return GulpUtils.streamToPromise(gulp.src(path.join(srcPath, "/*/package.json"))
+            .pipe(jsonEditor(
+                function(json: IPackageJson): IPackageJson {
+                    // update the version
+                    json.version = GulpUtils.transformPackageVersion(json.version, buildType);
+                    Object.keys(json.dependencies || {}).forEach(function(packageKey: string): void {
+                        if (tacoModules.indexOf(packageKey) >= 0) {
+                            json.dependencies[packageKey] = GulpUtils.getDependencyValue(packageKey, srcPath, buildType, destPath);
+                        }
+                    });
+                    Object.keys(json.optionalDependencies || {}).forEach(function(packageKey: string): void {
+                        if (tacoModules.indexOf(packageKey) >= 0) {
+                            json.optionalDependencies[packageKey] = GulpUtils.getDependencyValue(packageKey, srcPath, buildType, destPath);
+                        }
+                    });
+                    return json;
+                }))
+            .pipe(gulp.dest(destPath)));
+    }
+
+    private static updateDynamicDependencies(srcPath: string, buildType: string, destPath: string): Q.Promise<any> {
+        return GulpUtils.streamToPromise(gulp.src(path.join(srcPath, "/*/dynamicDependencies.json"))
+            .pipe(jsonEditor(
+                function(json: IDynamicDependenicesJson): IDynamicDependenicesJson {
+                    Object.keys(json).forEach(function(packageKey: string): void {
+                        var entry: IDynamicDependencyEntry = json[packageKey];
+                        if (entry.dev) {
+                            switch (buildType) {
+                                case "dev":
+                                    entry.localPath = GulpUtils.getDevDependencyValue(entry.packageName, srcPath, destPath);
+                                    break;
+                                case "release":
+                                    entry.dev = undefined;
+                                    entry.localPath = undefined;
+                                    entry.packageId = entry.packageName;
```

Comment:
This isn't quite right I think. It'll work for now, but in future if we start having e.g. taco-remote-lib@1.2.3 and taco-remote-lib@2.0.0 supporting different sets of functionality, we don't want to strip out that side of things. Even if we have frozen 1.2.3 so only 2.0.0 has the dev flag, we want to point to 2.0.0 specifically I believe.